src/tiled/tileselectiontool.cpp

```diff
@@ -190,8 +194,13 @@ void TileSelectionTool::modifiersChanged(Qt::KeyboardModifiers modifiers)
         mSelectionMode = Intersect;
         mIntersect->setChecked(true);
     } else {
-        mSelectionMode = Replace;
-        mReplace->setChecked(true);
+        mSelectionMode = mDefaultMode;
+        switch (mDefaultMode) {
```

Comment:
If you make this switch based on `mSelectionMode`, you can move it out of the condition and remove all the `setChecked(true)` lines above. Then, all the bodies are just one line so you can even remove all the brackets. :-)